src/server/scripts/EasternKingdoms/Stratholme/boss_baroness_anastari.cpp

```diff
@@ -16,66 +16,82 @@
  * with this program. If not, see <http://www.gnu.org/licenses/>.
  */
 
-/* ScriptData
-SDName: Boss_Baroness_Anastari
-SD%Complete: 90
-SDComment: MC disabled
-SDCategory: Stratholme
-EndScriptData */
-
-#include "ScriptMgr.h"
 #include "InstanceScript.h"
+#include "ObjectAccessor.h"
+#include "Player.h"
+#include "ScriptMgr.h"
 #include "ScriptedCreature.h"
 #include "stratholme.h"
 
 enum Spells
 {
-    SPELL_BANSHEEWAIL       = 16565,
-    SPELL_BANSHEECURSE      = 16867,
-    SPELL_SILENCE           = 18327
-    //SPELL_POSSESS           = 17244
+    SPELL_BANSHEEWAIL           = 16565,
+    SPELL_BANSHEECURSE          = 16867,
+    SPELL_SILENCE               = 18327,
+    SPELL_POSSESS               = 17244,
+    SPELL_POSSESSED             = 17246
+};
+
+enum BaronessAnastariEvents
+{
+    EVENT_SPELL_BANSHEEWAIL     = 1,
+    EVENT_SPELL_BANSHEECURSE    = 2,
+    EVENT_SPELL_SILENCE         = 3,
+    EVENT_SPELL_POSSESS         = 4,
+    EVENT_INVISIBLE             = 5,
+    EVENT_CHECK_POSSESSED       = 6
 };
 
 class boss_baroness_anastari : public CreatureScript
 {
 public:
     boss_baroness_anastari() : CreatureScript("boss_baroness_anastari") { }
 
-    CreatureAI* GetAI(Creature* creature) const override
-    {
-        return GetStratholmeAI<boss_baroness_anastariAI>(creature);
-    }
-
-    struct boss_baroness_anastariAI : public ScriptedAI
+    struct boss_baroness_anastariAI : public BossAI
     {
-        boss_baroness_anastariAI(Creature* creature) : ScriptedAI(creature)
+        boss_baroness_anastariAI(Creature* creature) : BossAI(creature, TYPE_BARONESS)
```

Comment:
I don't know if `TYPE_BARONESS` correct.

Reply:
`TYPE_BARONESS` refers to this line in `stratholme.h`:
https://github.com/TrinityCore/TrinityCore/blob/fb31cb7c047f4d79e7bea04b72175e6ef251a317/src/server/scripts/EasternKingdoms/Stratholme/stratholme.h#L27-L30
... meaning the value it represents is 2. Not sure if that is what you intended.